odoo/tools/float_utils.py

```diff
@@ -44,24 +51,42 @@ def _float_check_precision(precision_digits=None, precision_rounding=None):
         raise AssertionError(msg)
     return precision_rounding
 
-
-def float_round(value, precision_digits=None, precision_rounding=None, rounding_method='HALF-UP'):
+@overload
+def float_round(
+    value: float,
+    precision_digits: int,
+    rounding_method: RoundingMethod = ...,
+) -> float: ...
+
+@overload
+def float_round(
+    value: float,
+    precision_rounding: float,
+    rounding_method: RoundingMethod = ...,
+) -> float: ...
```

Comment:
This adds almost no value, and may even introduce confusion to devs that are not comfortable with Python types.  Please keep it simple, the base function signature is enough.

Reply:
With them, the IDEs are able to display a meaningful tooltip with the actual two use cases of this method:
<img width="1062" alt="Screenshot 2025-03-17 at 6 33 11 PM" src="https://github.com/user-attachments/assets/e0490182-af34-4667-8153-c2f3d585752e" />

Without it, the annotation is incomplete as it's not immediately clear that either one of `precision_rounding` and `rounding_method` should be provided, but not both at the same time or none of them:
<img width="1063" alt="Screenshot 2025-03-17 at 6 33 33 PM" src="https://github.com/user-attachments/assets/14689f25-2fc6-4f22-ad2e-db83df8b1ff0" />

I think it's useful, but I'll remove them if you still don't agree. Let me know